app/Progress.cs

```diff
@@ -0,0 +1,239 @@
+
+    //    db.Add(result);
+
+    //    db.SaveChanges();
+
+    //    return result;
+    //}
+
+
+
+    // Overloaded Execute Function with additional Arguments (Ex. Delete, Edit, Create) -> lms Progress Delete 0, lms Progress Edit 3
+    public void Execute(Verb verb, string[] command_args)
+    {
+        switch (verb) 
+        {
+            case Verb.List:
+                Execute(verb);
+                break;
+            case Verb.Delete:
+                if (command_args.Count() < 1) {
+                    throw new ArgumentException("Delete requires an ID!");
+                }
+
+                var delete_result = Delete(command_args[0]);
+
+                Console.WriteLine("----------------");
+                Console.WriteLine($"ID: {delete_result.Id}");
+                Console.WriteLine($"Description: {delete_result.Description}");
+                Console.WriteLine($"WorkItem: {delete_result.WorkItem.Id}");
+                Console.WriteLine($"CreatedAt: {delete_result.CreatedAt}");
+                Console.WriteLine("----------------");
+                break;
+            case Verb.Edit:
+                if (command_args.Count() < 3) {
+                    throw new ArgumentException("3 arguments: Id, Field, and Value!");
+                }
+
+                var edit_result = Edit(command_args[0], command_args[1], command_args[2]);
+
+                Console.WriteLine("----------------");
+                Console.WriteLine($"ID: {edit_result.Id}");
+                Console.WriteLine($"Description: {edit_result.Description}");
+                Console.WriteLine($"WorkItem: {edit_result.WorkItem.Id}");
+                Console.WriteLine($"CreatedAt: {edit_result.CreatedAt}");
+                Console.WriteLine("----------------");
+
+                break;
+            // Not Impletemented Yet
+
+            // WorkItem Create case Implementation for Reference
+            //case Verb.Create:
+            //    if (command_args.Count() < 1)
+            //    {
+            //        throw new ArgumentException("Create requires at least a Title Arg");
+            //    }
+
+            //    var title = command_args[0];
+            //    string? due_at = command_args.ElementAtOrDefault(1);
+
+            //    var create_result = Create(title, due_at);
+
+            //    Console.WriteLine("----------------------");
+            //    Console.WriteLine($"Id: {create_result.Id}");
+            //    Console.WriteLine($"Title: {create_result.Title}");
+            //    Console.WriteLine($"CreatedAt: {create_result.CreatedAt}");
+            //    Console.WriteLine($"DueAt: {create_result.DueAt}");
+            //    break;
+                
+            default:
+                throw new ArgumentException("Invalid Verb");
+        }
+    }
+
+    // Edit Progress
+    public Lms.Models.Progress Edit(string id, string field, string value) {
+        int parsed_id = -1;
```

Comment:
There's a concept called [Resource Acquisition is Initialization](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization) which basically is the idea to set variable values at time of declaration (to avoid things like having to think of temp values).

I think it is A OK here :) just thought to share this in case you're interested!